web/pages/api/v0/increment.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  CORS_ORIGIN_MANIFOLD,
+  CORS_ORIGIN_LOCALHOST,
+} from 'common/envs/constants'
+import { NextApiRequest, NextApiResponse } from 'next'
+import { applyCorsHeaders } from 'web/lib/api/cors'
+import * as admin from 'firebase-admin'
+import { auth } from 'web/lib/firebase/users'
+
+export const config = { api: { bodyParser: true } }
+
+function initAdmin() {
+  // TODO: Support dev vs prod environments; maybe private instances too?
+  // TODO: How do we get the Firebase Admin secret to exist here, _without_ leaking it
+  // to PR preview branches from our users? (Maybe Vercel already does this safely?)
+  const serviceAccount = JSON.parse(
+    process.env.FIREBASE_SERVICE_ACCOUNT_KEY ?? ''
+  )
+  if (admin.apps.length === 0) {
+    admin.initializeApp({
+      credential: admin.credential.cert(serviceAccount),
+    })
+  }
+}
+initAdmin()
+
+async function getUserId(req: NextApiRequest, res: NextApiResponse) {
+  // Check for the correct bearer token. Taken from functions/src/api.ts
+  const authHeader = req.headers.authorization ?? ''
+  const authParts = authHeader.split(' ')
+  if (authParts.length !== 2) {
+    res.status(403).json({ error: 'Invalid Authorization header.' })
+  }
+  const [scheme, payload] = authParts
+  if (scheme !== 'Bearer') {
+    // TODO: Support API Key too
+    res.status(403).json({ error: 'Invalid auth scheme; must be "Bearer".' })
+  }
+  // Seems to involve a roundtrip to Firebase; could we skip? (Or locate Vercel server such that this is fast?)
+  const decodedToken = await admin.auth().verifyIdToken(payload)
```

Comment:
The round trip isn't supposed to happen every time; it's supposed to cache the public keys in memory. So you can go add this to the giant list of serverless function complaints I posted in Discord if you want.